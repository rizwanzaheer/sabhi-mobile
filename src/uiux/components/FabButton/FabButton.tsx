import * as React from 'react'
import { TouchableOpacity } from 'react-native'
import Icon, { IconConfig } from '../Icon/Icon'
import { withTheme } from '../../theming'

interface FabButtonProps {
  /**
   * Test ID
   */
  testID?: string

  /**
   * Button color. Defaults to brand color
   */
  color?: string

  /**
   * Shadow depth. Defaults to .3
   */
  shadowOpacity?: number

  /**
   * Icon to show
   */
  icon?: IconConfig

  /**
   * Color of icon. Defaults to primary background
   */
  iconColor?: string

  /**
   * Handler
   */
  onPress: () => void

  /**
   * Custom size
   */
  size?: number

  /**
   * Theme
   */
  theme: any
}

const FabButton: React.FC<FabButtonProps> = props => {
  const shadowStyle = {
    shadowColor: '#000000',
    shadowOpacity: props.shadowOpacity ? props.shadowOpacity : 0.3,
    shadowRadius: 20,
    elevation: props.shadowOpacity ? props.shadowOpacity * 10 : 3,
  }

  return (
    <TouchableOpacity
      accessibilityLabel={props.testID}
      testID={props.testID}
      onPress={props.onPress}
      style={{
        height: props.size || 70,
        width: props.size || 70,
        backgroundColor: props.color || props.theme.colors.primary.brand,
        borderRadius: 50,
        alignItems: 'center',
        justifyContent: 'center',
        ...shadowStyle,
      }}
    >
      {props.icon && (
        <Icon icon={props.icon} size={35} color={props.iconColor || props.theme.colors.primary.background} />
      )}
    </TouchableOpacity>
  )
}

export default withTheme(FabButton)
