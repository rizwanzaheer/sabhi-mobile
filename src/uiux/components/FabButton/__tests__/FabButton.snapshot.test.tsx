import * as React from 'react'
import { render } from 'react-native-testing-library'
import FabButton from '../FabButton'

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')

describe('Component(snapshots): FabButton', () => {
  it('should render with required props', () => {
    const onPress = jest.fn()
    const tree = render(<FabButton onPress={onPress} />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with all props', () => {
    const onPress = jest.fn()
    const tree = render(
      <FabButton
        shadowOpacity={0.2}
        onPress={onPress}
        icon={{ name: 'ios-add', iconFamily: 'Ionicons' }}
        iconColor={'#000000'}
      />,
    )
    expect(tree).toMatchSnapshot()
  })

  it('should render with default color', () => {
    const onPress = jest.fn()
    const tree = render(<FabButton onPress={onPress} icon={{ name: 'ios-add', iconFamily: 'Ionicons' }} />)
    expect(tree).toMatchSnapshot()
  })
})
