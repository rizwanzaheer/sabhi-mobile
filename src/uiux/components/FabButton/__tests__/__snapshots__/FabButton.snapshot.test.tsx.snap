// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): FabButton should render with all props 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0078ff",
      "borderRadius": 50,
      "elevation": 2,
      "height": 70,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOpacity": 0.2,
      "shadowRadius": 20,
      "width": 70,
    }
  }
>
  <Icon
    color="#000000"
    name="ios-add"
    size={35}
  />
</View>
`;

exports[`Component(snapshots): FabButton should render with default color 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0078ff",
      "borderRadius": 50,
      "elevation": 3,
      "height": 70,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOpacity": 0.3,
      "shadowRadius": 20,
      "width": 70,
    }
  }
>
  <Icon
    color="#FFFFFF"
    name="ios-add"
    size={35}
  />
</View>
`;

exports[`Component(snapshots): FabButton should render with required props 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0078ff",
      "borderRadius": 50,
      "elevation": 3,
      "height": 70,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOpacity": 0.3,
      "shadowRadius": 20,
      "width": 70,
    }
  }
/>
`;
